SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,952 @@
+                string[] options = Enum.GetNames(memberInfoMetadata.ValueType);
+                string value = memberInfoMetadata.GetValue(Config).ToString();
+                AddChoiceOption(id, label, options, value);
+            }
+            else if (memberInfoMetadata.ValueType.IsEnum)
+            {
+                // Enum-based choice where the values are defined as custom strings
+                string[] options = choiceAttribute.Options;
+                string value = memberInfoMetadata.GetValue(Config).ToString();
+                int index = Array.IndexOf(Enum.GetValues(memberInfoMetadata.ValueType), value);
+                AddChoiceOption(id, label, options, index);
+            }
+            else if (memberInfoMetadata.ValueType == typeof(string))
+            {
+                // string-based choice value
+                string[] options = choiceAttribute.Options;
+                string value = memberInfoMetadata.GetValue<string>(Config);
+                AddChoiceOption(id, label, options, value);
+            }
+            else if (memberInfoMetadata.ValueType == typeof(int))
+            {
+                // index-based choice value
+                string[] options = choiceAttribute.Options;
+                int index = memberInfoMetadata.GetValue<int>(Config);
+                AddChoiceOption(id, label, options, index);
+            }
+        }
+
+        /// <summary>
+        /// Adds a <see cref="ModKeybindOption"/> to the <see cref="ModOptions"/> menu.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="label"></param>
+        /// <param name="memberInfoMetadata">The metadata of the corresponding member.</param>
+        private void BuildModKeybindOption(string id, string label, MemberInfoMetadata memberInfoMetadata)
+        {
+            var value = memberInfoMetadata.GetValue<KeyCode>(Config);
+            AddKeybindOption(id, label, GameInput.Device.Keyboard, value);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="ModSliderOption"/> to the <see cref="ModOptions"/> menu.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="label"></param>
+        /// <param name="memberInfoMetadata">The metadata of the corresponding member.</param>
+        /// <param name="sliderAttribute">The defined or generated <see cref="SliderAttribute"/> of the member.</param>
+        private void BuildModSliderOption(string id, string label,
+            MemberInfoMetadata memberInfoMetadata, SliderAttribute sliderAttribute)
+        {
+            float value;
+            if (memberInfoMetadata.ValueType == typeof(float))
+            {
+                value = memberInfoMetadata.GetValue<float>(Config);
+            }
+            else if (memberInfoMetadata.ValueType == typeof(double))
+            {
+                value = Convert.ToSingle(memberInfoMetadata.GetValue<double>(Config));
```

Comment:
Since I'm the one who wrote the `GetValue` I am very aware that I have a version that works with `object`, and I do also know that `Convert.ToSingle` works with `object`. Please start assuming that there is a reason for my choices other than my just not knowing what I am doing, it's getting rather irksome.

The types are checked in `ProcessFieldOrProperty`, yes, so the chances of it falling apart when we reach this piece of code are impossible, so yes, technically it would be okay to call `Convert.ToSingle` on an `object`. However, this is more about bulletproofing it against future changes to the codebase. The logic is easier to deciper for someone else reading than if you're just working with an `object`. Furthermore, since there's no need to call `Convert.ToSingle` when we know the type is a float, I'd rather just compare the types.

Reply:
Well, I think that
```
float value = Convert.ToSingle(memberInfoMetadata.GetValue(Config));
```
is easier to read and understand than all those lines, but maybe I'm wrong. If that was an intentional decision, ok then.

Reply:
@PrimeSonic when you have a chance, I'd ask you to weigh in on this.

Reply:
Resolved by c26bcdd